presto-main/src/main/java/com/facebook/presto/metadata/Signature.java

```diff
@@ -40,6 +41,8 @@
     private final TypeSignature returnType;
     private final List<TypeSignature> argumentTypes;
     private final boolean variableArity;
+    private final Optional<String> catalog;
```

Comment:
I don't think it's valid to have a schema without a catalog. This should probably be `Optional<CatalogSchemaName>`